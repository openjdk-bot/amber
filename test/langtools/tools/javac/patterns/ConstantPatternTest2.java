/*
 * @test
 * @summary Basic tests for constant patterns
 * @compile/fail/ref=ConstantPatternTest2.out -XDrawDiagnostics --enable-preview -source ${jdk.version} ConstantPatternTest2.java
 */

public class ConstantPatternTest2 {

    public static double valueOfD(){
        return 41.0;
    }

    public static void main(String[] args) {

        int i = 41;
        i++;
        final int constantInt = 42;


        // Simple literal constant pattern
        if (i instanceof 42) {
            throw new AssertionError("Broken");
        }

        // Simple constant expression pattern
        if (i instanceof 41+1) {
            throw new AssertionError("Broken");
        }

        // Constant expression pattern using a final local
        if (i instanceof constantInt) {
            throw new AssertionError("Broken");
        }

//--- DOUBLE TYPED CONSTANT EXPRESSIONS

        double d = valueOfD();
        d++;
        final double constant = 42.0;

        // Simple literal constant pattern
        if (d instanceof 42.0) {
            throw new AssertionError("Broken");
        }

        // Simple constant expression pattern
        if (d instanceof 41.0+1.0) {
            throw new AssertionError("Broken");
        }

        // Constant expression pattern using a final local
        if (d instanceof constant) {
            throw new AssertionError("Broken");
        }

//--- STRING TYPED CONSTANT EXPRESSIONS ----

        String s = "Hello";
        final String hello = "Hello";

        // Simple literal constant pattern
        if (s instanceof "Hello") {
            throw new AssertionError("Broken");
        }

        // Simple constant expression pattern
        if (s instanceof "Hell"+"o") {
            throw new AssertionError("Broken");
        }

        // Constant expression pattern using a final local
        if (s instanceof hello) {
            throw new AssertionError("Broken");
        }

// ----- Constant expression patterns and reference typed expressions

        Object obj = "Hello";

        if (obj instanceof "Hello") {
            throw new AssertionError("Broken");
        }
        if (obj instanceof null) {
            throw new AssertionError("Broken");
        }
        obj = 42;
        if (obj instanceof 42) {
            throw new AssertionError("Broken");
        }

        if (obj instanceof var x) {
            throw new AssertionError("Broken");
        }

        if (obj instanceof 42) {
        }
        if (obj instanceof null) {
            throw new AssertionError("Broken");
        }
        if (null instanceof "Hello") {
            throw new AssertionError("Broken");
        }

    }

}
