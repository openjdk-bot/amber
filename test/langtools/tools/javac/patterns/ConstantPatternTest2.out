ConstantPatternTest2.java:21:26: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:21:28: compiler.err.expected: ';'
ConstantPatternTest2.java:26:26: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:26:28: compiler.err.not.stmt
ConstantPatternTest2.java:26:30: compiler.err.expected: ';'
ConstantPatternTest2.java:42:26: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:42:30: compiler.err.expected: ';'
ConstantPatternTest2.java:47:26: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:47:30: compiler.err.not.stmt
ConstantPatternTest2.java:47:34: compiler.err.expected: ';'
ConstantPatternTest2.java:62:26: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:62:33: compiler.err.expected: ';'
ConstantPatternTest2.java:67:26: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:67:36: compiler.err.expected: ';'
ConstantPatternTest2.java:80:28: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:80:35: compiler.err.expected: ';'
ConstantPatternTest2.java:83:28: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:83:32: compiler.err.expected: ';'
ConstantPatternTest2.java:87:28: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:87:30: compiler.err.expected: ';'
ConstantPatternTest2.java:91:17: compiler.err.restricted.type.not.allowed: var, 10
ConstantPatternTest2.java:95:28: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:95:30: compiler.err.expected: ';'
ConstantPatternTest2.java:97:28: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:97:32: compiler.err.expected: ';'
ConstantPatternTest2.java:100:29: compiler.err.illegal.start.of.type
ConstantPatternTest2.java:100:36: compiler.err.expected: ';'
- compiler.note.preview.filename: ConstantPatternTest2.java
- compiler.note.preview.recompile
27 errors
